### DISCUSSION TITLE
How can you run the generation tool for fixed questions?

### DISCUSSION THREAD

**Discussion by nheidloff** [instructlab/instructlab#1040] [Q&A]
Hi, I like the generation capability. Context and answers should be generated, but in my case I want to keep the question fixed. Is this possible?

**Comment by wking** [comment]
None of [the current `generate` options][1] allow you to pin the questions.  I'm not LLM-aware enough to know if pinning a question would be useful (lots of generated examples helping the model learn how to answer that exact question!) or not (sometimes folks might ask similar, but not identical, questions, and we want your training to help them too).

[1]: https://github.com/instructlab/instructlab/blob/503bc133c311702b023ccb976a654e5ad49f988e/src/instructlab/lab.py#L335-L449

**Reply by nheidloff** [reply]
I think pinning the question is quite common. Since the context is different, the model can learn.

Text extraction is a good example. The question/instruction is always the same one.

When I let the generation script create data, the questions often don't make sense. And since the generated bad questions are used to generate answers, the answers are even worse.
